### ISSUE TITLE
Power menu not working properly 

### ISSUE THREAD

**Issue by TheOne988** [realcopacetic/skin.copacetic#39]
There is a problem with the Power menu. Basically the only thing that works on the power menu is the Light and Dark mode. Any other item that is added to the power menu won't open. Even the defualt Add-ons one is not opening.

I have tried to open Add-ons, System information and System till now but without luck. The test ware performed when opening the power menu from the home screen also on opened settings menu.

I use LibreElec 11.0.1 and the problem is present on the latest skin 1.1.6.

**Comment by realcopacetic** [comment]
By chance if you look in your log, do you see this error? 

"Activate of window '10025' refused because there are active modal dialogs"

I saw someone else having this issue on Copacetic and also on Estuary as well, so it could well be an issue with the dependency skin.shortcuts.

Which version of it do you have? 

In addition to the Night/Light toggle, you should have a hardcoded settings shortcut in the power menu as long as you're not in a settings screen. And a home shortcut as long as you're not on the home screen. Do these work?

**Comment by TheOne988** [comment]
Hi,

yes that is the exact info I get:

`info <general>: Activate of window '10040' refused because there are active modal dialogs`

I'm using skin.shortcuts v 2.0.3 and have tried to download id and install it from a ZIP file but no luck.

Anyway this behavior is Linux related most defensively. I have just installed Kodi on Windows and everything works as it should. I get this kind of behavior only on LibreElec and Kodi (flatpak) on clean install no other modifications or installation is done before installing Copacetic Skin.

I also can confirm that skin.shortcuts is working correctly with other skins for example Arctic: Zephyr - Reloaded
![screenshot00001](https://github.com/realcopacetic/skin.copacetic/assets/36518440/075e1330-8898-4610-bcbf-a8c26c2152bb)

About the hard coded menu. The Night/Light works as it should from home screen and from the settings screen. But the Home menu that is available everywhere except the home screen is not working (I cannot see any errors in the kodi.log). Basically when I press the Home hard coded button i get to a grayed screen with nothing in it and cant get out of it unless I restart kodi or reboot the system (also no visible errors in the kodi.log).

![screenshot00002](https://github.com/realcopacetic/skin.copacetic/assets/36518440/95143dcb-bba6-43c1-bc1a-2c62764c42db)

Can you advice what else I should try or do you need any additional info regarding this error.

**Comment by realcopacetic** [comment]
Ok if it's not happening in other skins, it just be something in how copacetic is using skin shortcuts. I'll try and investigate further thanks for the extra info

**Comment by realcopacetic** [comment]
I'm a bit stumped at the moment. Could you share your log after opening kodi and trying a skin shortcut that you added plus the Home shortcut.

I really don't know why the hardcoded shortcuts wouldn't work as they are specifically closing all other windows then activating for e.g. home

```
<item>
    <description>Home</description>
    <label>$LOCALIZE[31401]</label>
    <onclick>Dialog.Close(all)</onclick>
    <onclick>ActivateWindow(home)</onclick>
    <visible>!Window.IsVisible(home) + !Window.IsVisible(loginscreen)</visible>
</item>
```

It's really strange

**Comment by realcopacetic** [comment]
Is it same for the Settings shortcut in the power menu as well? The hardcoded items are the theme switcher, Home and Settings. You should see settings whenever you're not in a settings screen. Does that work?

**Comment by realcopacetic** [comment]
Actually one thing you can try. If you don't mind going to your addon folder and editing a value in one file for Copacetic with a text editor to turn on debugging.

addons/skin.copacetic/addon.xml > somewhere it will say debugging="false" if you turn that to "true" and save. Then you can open Kodi and you should see some info in the top left about the currently active window and control. If you use that home shortcut in the powermenu, what does it say in the top left? I'm wondering if the Home window has actually loaded but for some reason is empty

**Comment by TheOne988** [comment]
Hi,

I will answer you point by point like you have written the comments.

- first comment:
you can find two log files attached (kody_system.log and kodi_home.log)
- the kodi_system.log is a LOG file from starting kodi opening the shutdown menu by pressing the S key and selecting System (the added additional menu was system and it sould have oppened "ActivateWindows(SystemSettings)") afterwards I have opened the shutdown menu again and clicked exit.
- the kodi_home.log is a LOG file from starting kodi going to the settings menu and then opening.  the shutdown menu by pressing S key and klicking Home. After this the windows is just black.

[kodi_system.log](https://github.com/realcopacetic/skin.copacetic/files/12241542/kodi_system.log)
[kodi_home.log](https://github.com/realcopacetic/skin.copacetic/files/12241544/kodi_home.log)

- seccond comment:
Yes I can open the settings from the shutdown menu when I'm in the home screen. The settings menu is not available when I'm already in the settings menu.

- third comment:
when I enabled debuging here are the results
-- on the home screen I see: Window: home (Home.xml)
-- when I press S to open the shutdown menu I see: Window: shutdown (DialogButtonMenu.xml)
-- when I try to open the System settings (custom added shortcut) from the shutdown menu the message stays same as above ( Window: shutdown (DialogButtonMenu.xml)

**Comment by TheOne988** [comment]
Additionally I have noticed when I press the "Home" button from the shutdown menu I get:
Window: home (Home.xml)
Mouse: (980, 726) Focused 4010 (button)

But when I am really on the home screen I see:
Window: home (Home.xml)
Mouse: (980, 726) Focused 3000 (list)

So basically it does not show the home menu list...

**Comment by TheOne988** [comment]
Here is a short video of all the problems I'm describing with enabled debugging in addon.xml
Additionally at the moment I have no Video, TV Shows... menu because I did a clean install thinking that could solve my issue. 
 
https://github.com/realcopacetic/skin.copacetic/assets/36518440/93139743-1936-4bb8-8dcf-978a118d093a

**Comment by realcopacetic** [comment]
Thanks @TheOne988 - Can you try installing 1.1.7 from my repository and see if that fixes any of these issues please

**Comment by TheOne988** [comment]
Hi @realcopacetic,

I have tried upgrading the skin via your repo and doing a clean install just via the repo and I can confirm that the "HOME" button is now working like it should, unfortunately anything else is still the same.

I still get the same error from the shutdown menu for "Add-ons" or any custom created button.
`info <general>: Activate of window '10016' refused because there are active modal dialogs`
`info <general>: Activate of window '10040' refused because there are active modal dialogs`

P.S. you have a typo in the Readme.md file in the Changelog your version number should be 1.1.7 not 1.1.6 😊

By the way thanks a lot for your effort and trying to fix this.

**Comment by realcopacetic** [comment]
It's strange. I can replicate the issue you're having by removing the following lines from the file: [skin.copacetic/shortcuts/Overrides.xml](https://github.com/realcopacetic/skin.copacetic/blob/master/shortcuts/Overrides.xml)

`<backgroundBrowse>True</backgroundBrowse>`
`<background label="::PLAYLIST::">playlistBackground</background>`
`<groupoverride group="shutdownmenu" condition="Window.IsActive(DialogButtonMenu.xml)">Dialog.Close(all,force)</groupoverride>`

It's almost as if that file isn't loading properly for some reason. Out of interest, if you change the filename to a small 'o' does that have any impact? I know some things can be case sensitive on Linux and that's the only other difference I can see with the same file in Arctic Zephyr Reloaded

**Comment by TheOne988** [comment]
Hi @realcopacetic,

I can confirm that changing the Override.xml file to override.xml fixes all the issues.
It fixes completely the shutdown menu and the main menu with the background options missing.

At the beginning I was also comparing the Arctic Zephyr Reloaded configuration files with yours and couldn't find anything significantly different that could cause a problem.

It turns out it was that simple 😀

Anyway just a note because I changed the filename on a configured skin I needed to recreate the shutdown menu in order to start working.

Thanks a lot for the FIX I can now finally start and using the awesome skin.
